importer/src/module_loader.rs

```diff
@@ -25,6 +65,47 @@ type Error = String;
 impl<T: FieldElement> Folder<T> for Loader {
     type Error = Error;
 
+    fn fold_module_value(&mut self, module: ASMModule<T>) -> Result<ASMModule<T>, Self::Error> {
+        // Fold module statements. This code is identical to the default implementation of Folder::fold_module_value
```

Comment:
Can't you call it using `Folder::fold_module_value` from here?

Reply:
Didn't find a way, unfortunately. Calling `Folder::fold_module_value` seems to just call `Loader::fold_module_value`, leading to infinite recursion. I also tried doing this:
```rust
struct AuxFolder();
impl<T: FieldElement> Folder<T> for Loader {
    type Error = Error;
}

impl<T: FieldElement> Folder<T> for Loader {
    type Error = Error;

    fn fold_module_value(&mut self, module: ASMModule<T>) -> Result<ASMModule<T>, Self::Error> {
        let module = AuxFolder().fold_module_value(module)?;
        // ...
    }
}
```

This calls the right function, but then also calls `AuxFolder::fold_module` (the default implementation) down the call stack, so external imports are not resolved...

Reply:
This is obsolete now as well, as these are two separate passes now.